migen/genlib/fifo.py

```diff
@@ -182,8 +194,9 @@ class AsyncFIFO(Module, _FIFOInterface):
     """
     __doc__ = __doc__.format(interface=_FIFOInterface.__doc__)
 
-    def __init__(self, width, depth):
-        _FIFOInterface.__init__(self, width, depth)
+    def __init__(self, width, depth, fwft=False):
+        assert not fwft, "fwft is not supported on the AsyncFIFO."
+        _FIFOInterface.__init__(self, width, depth, fwft=False)
```

Comment:
This ``AsyncFIFO`` is always FWFT (in general there is no reason for an async FIFO not to be FWFT).

Reply:
With a sync FIFO it's either:
* non-FWFT
* asynchronous reads (lots of resources used, timing issues)
* 2 cycles of latency instead of 1

The async FIFO does not have to make this compromise (the latency is already high enough from the CDC and hides the memory latency).